swap/src/main.rs

```diff
@@ -249,10 +243,5 @@ async fn setup_wallets(
     );
     let monero_wallet = Arc::new(monero_wallet);
```

Comment:
I would defer `Arc`ing things up until the point where you need it (i.e. to the first call of `clone`). This function seems so know too much about what is happening afterwards :)